cedar-policy-validator/src/coreschema.rs

```diff
@@ -67,13 +64,17 @@ impl<'a> entities::Schema for CoreSchema<'a> {
         &'b self,
         basename: &'b ast::Id,
     ) -> Box<dyn Iterator<Item = ast::EntityType> + 'b> {
-        Box::new(self.schema.entity_types().filter_map(move |(name, _)| {
-            if name.basename() == basename {
-                Some(ast::EntityType::Specified(name.clone()))
-            } else {
-                None
-            }
-        }))
+        Box::new(
+            self.schema
+                .entity_types()
+                .filter_map(move |(entity_type, _)| {
+                    if entity_type.name().basename() == basename {
+                        Some(ast::EntityType::from(entity_type.name().clone()))
```

Comment:
```suggestion
                        Some(entity_type)
```
(Clone may still be needed)